src/mainwindow.cpp

```diff
@@ -268,13 +288,38 @@ QString MainWindow::saveFileName(QString defaultName) {
   return fileName;
 }
 
+QString MainWindow::firstImageOfDir(QString dirname)
+{
+    QList<QByteArray> formatsBytes = QImageReader::supportedImageFormats();
+    QStringList formats;
+    for (const QByteArray& formatBytes: formatsBytes) {
+        formats << QString("*.") + formatBytes;
+    }
+    QDir dir(dirname);
+    dir.setNameFilters(formats);
+    dir.setFilter(QDir::Files | QDir::NoDotAndDotDot);
+    dir.setSorting(QDir::Name);
+    QList<QFileInfo> imageFiles = dir.entryInfoList();
+    if (!imageFiles.empty()) {
+        return imageFiles.first().absoluteFilePath();
+    }
+    return QString();
+}
+
```

Comment:
Unfortunately, `MainWindow::firstImageOfDir()` doesn't necessarily give the first image that can be seen in the thumbnail view. For example, if a folder contains "3.jpg", "7.jpg", "101.jpg" and "411.jpg", this function gives  "101.jpg" while "3.jpg" is the first image in the thumbnail view.